nbgitpuller/__init__.py

```diff
@@ -4,6 +4,16 @@
 from notebook.utils import url_path_join
 from tornado.web import StaticFileHandler
 import os
+import nest_asyncio
+
+REPO_PARENT_DIR = None
```

Comment:
This was previously decided in the SyncHandler's get function, but since it is now needed by the `handle_files_helper` in `plugin_helper.py`, it has now been declared as a variable in __init__.py and set from the SyncHandler's get function to then be imported in `plugin_helper.py` and used by `handle_files_helper`.

I think this builds more complexity than is sustainable.

If this is needed to be defined at a level common to the SyncHandler and the `handle_files_helper`, I would expect it to be calculated at that level rather than rely on being passed back from a lower level when that is executed for the first (and consecutive times).

This is the logic from the SyncHandler's get function:

```python
            repo_parent_dir = os.path.join(os.path.expanduser(self.settings['server_root_dir']), os.getenv('NBGITPULLER_PARENTPATH', ''))
            nbgitpuller.REPO_PARENT_DIR = repo_parent_dir
```

Some user is referenced and expanded, will that user be unique in the scope of __init__.py, or is this value also expected to change over time? Hmmm...

Is it possible to refactor this to reduce complexity, or document it to be clearer?

Reply:
yes, I can get it out of there. I think I follow you here. I had assumed that it was unique in the scope of the user's "server" but I may not quite be grasping how JupyterHub spawns instances and shares resources.